storage/objects/objects_test.go

```diff
@@ -256,6 +259,81 @@ func TestV4SignedURL(t *testing.T) {
 	}
 }
 
+func TestPostPolicyV4(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	bucketName := tc.ProjectID + "-post-policy-bucket-name"
+	objectName := "foo.txt"
+	serviceAccount := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+
+	cleanBucket(t, ctx, client, tc.ProjectID, bucketName)
+	putBuf := new(bytes.Buffer)
+	policy, err := generateSignedPostPolicyV4(putBuf, bucketName, objectName, serviceAccount)
+	if err != nil {
+		t.Fatalf("generateSignedPostPolicyV4: %v", err)
+	}
+	got := putBuf.String()
+	if want := "<form action="; !strings.HasPrefix(got, want) {
+		t.Errorf("got output %q, should start with %q", got, want)
+	}
+
+	// Now create a form using the returned post policy.
```

Comment:
I'm not sure I understand the rest of the test. Can you clarify or maybe add additional comments?

Reply:
The signed post policy allows an unauthenticated client to make a POST using the form values from policy.Fields. The remainder of the test is verifying that this actually works as expected against the live service, using the code from the sample. I can add a bit more of that explanation as a comment.